lib/util.js

```diff
@@ -205,15 +205,37 @@ function addBorderNode(g, prefix, rank, order) {
   return addDummyNode(g, "border", node, prefix);
 }
 
+function splitToChunks(array, chunkSize = CHUNKING_THRESHOLD) {
```

Comment:
just for readibility i would move these functions after maxRank


applyWithChunking first, followed by splitToChunks.

since we are not using `const myFn = () => {}` we can get away with this and makes it easier to read